src/components/TableSubComponents/ViewDropdown/ViewItemPropTypes.js

```diff
@@ -0,0 +1,10 @@
+import PropTypes from 'prop-types';
+
+const ViewItemPropType = PropTypes.shape({
+  id: PropTypes.string.isRequired,
+  text: PropTypes.string.isRequired,
+  public: PropTypes.bool,
+  customAction: PropTypes.func,
```

Comment:
it would be good to document what this function does and the args.  ie, customAction, if used will override the default changeView action and changeView is never called if you use a customAction.